### PR TITLE
cmake: kconfig: Remove unused extra dtc flags

### LINKED ISSUE #33693: cmake -E env: unknown option '-Wno-unique_unit_address_if_enabled'
**Describe the bug**
The #31960 introduced the **pre_dt_board.cmake** file in many boards. The new rules add _EXTRA_DTC_FLAGS "-Wno-unique_unit_address_if_enabled"_ to build system. This rules if not proper defined will break _make menuconfig_.

**To Reproduce**
1. cd samples/hello_world
2. west build -b atsamr21_xpro
3. west build -t menuconfig

**Expected behavior**
**west build -t menuconfig** should never be broken by any change in Zephyr configuration.

**Impact**
It is not possible configure Zephyr without rollback.

**Logs and console output**
```
-- west build: running target menuconfig
...
cmake -E env: unknown option '-Wno-unique_unit_address_if_enabled'
...
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK sdk-0.12.3
 - Commit #mainline

**Additional context**
It is expected a test that can check broken Kconfig options

### PR DESCRIPTION
**Pull request by chrta**
If ninja was used and e.g. menuconfig was called, when more than one
element is in EXTRA_DTC_FLAGS, calling "ninja menuconfig" failed.

E.g.
west build -b lpcxpresso55s28_ns -t menuconfig samples/hello_world
resulted in an error. This is fixed with this commit.

Signed-off-by: Christian Taedcke <christian.taedcke@lemonbeat.com>

### CODE REVIEW

File: cmake/kconfig.cmake [hunk 1]
```diff
@@ -138,7 +138,7 @@ foreach(kconfig_target
     ZEPHYR_TOOLCHAIN_VARIANT=${ZEPHYR_TOOLCHAIN_VARIANT}
     ${COMMON_KCONFIG_ENV_SETTINGS}
     "SHIELD_AS_LIST=${SHIELD_AS_LIST_ESCAPED}"
-    EXTRA_DTC_FLAGS=${EXTRA_DTC_FLAGS}
```
Comment (tejlmand) [inline cmake/kconfig.cmake:141]: I believe this line can be completely removed.
It was originally introduced here https://github.com/zephyrproject-rtos/zephyr/commit/9aefdaff2ef893e303a67ae61f5ed3789d02ee61in order to share `EXTRA_DTC_FLAGS` with kconfigfunctions.py, but that need was removed later here https://github.com/zephyrproject-rtos/zephyr/pull/26589/commits/677b59a86dd0269dfc3a5d24cabc71856fbe5213 when edt is loaded from the file.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Thanks for this.
I think `EXTRA_DTC_FLAGS` can simply be removed as it doesn't appear to be needed anymore.

File: cmake/kconfig.cmake (hunk 1 above)
Comment (galak) [reply cmake/kconfig.cmake:141]: Would like @mbolivar-nordic to chime in on that.

Comment (mbolivar-nordic) [reply cmake/kconfig.cmake:141]: That looks right to me.

Comment (chrta) [reply cmake/kconfig.cmake:141]: So should i remove the line containing EXTRA_DTC_FLAGS here completely or should i leave it as it is now?

Comment (tejlmand) [reply cmake/kconfig.cmake:141]: Should be removed completely.

Comment (chrta) [reply cmake/kconfig.cmake:141]: done

**Review by tejlmand — APPROVED** [review]

**Comment by tejlmand** [comment]
Thanks for this and for the patience.

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**